flutter/lib/src/navigation/time_to_display_tracker.dart

```diff
@@ -0,0 +1,153 @@
+// ignore_for_file: invalid_use_of_internal_member
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+import '../integrations/integrations.dart';
+
+import '../../sentry_flutter.dart';
+import '../native/sentry_native.dart';
+import 'time_to_initial_display_tracker.dart';
+
+@internal
+class TimeToDisplayTracker {
+  final Hub? _hub;
+  final bool? _enableAutoTransactions;
+  final Duration? _autoFinishAfter;
+  final SentryNative? _native;
+  final TimeToInitialDisplayTracker _ttidTracker;
+
+  // TODO We can use _hub.options to fetch the ttfd flag
+  bool get _enableTimeToFullDisplayTracing => false;
+
+  TimeToDisplayTracker({
+    required Hub? hub,
+    required bool? enableAutoTransactions,
+    required Duration? autoFinishAfter,
+    TimeToInitialDisplayTracker? ttidTracker,
+  })  : _native = SentryFlutter.native,
+        _hub = hub ?? HubAdapter(),
+        _enableAutoTransactions = enableAutoTransactions,
+        _autoFinishAfter = autoFinishAfter,
+        _ttidTracker = ttidTracker ?? TimeToInitialDisplayTracker();
+
+  Future<ISentrySpan?> _startTransaction(String? routeName, Object? arguments,
+      {DateTime? startTimestamp}) async {
+    if (_enableAutoTransactions == false) {
+      return null;
+    }
+
+    if (routeName == null) {
+      return null;
+    }
+
+    final transactionContext = SentryTransactionContext(
+      routeName,
+      SentrySpanOperations.uiLoad,
+      transactionNameSource: SentryTransactionNameSource.component,
+      origin: SentryTraceOrigins.autoNavigationRouteObserver,
+    );
+
+    final transaction = _hub?.startTransactionWithContext(
```

Comment:
`m`: It seems a bit weird to me that the time to display tracker now starts the transaction. I think we should leave this responsibility to the navigation observer. Why did you choose this approach?